### PR TITLE
Add repeat_dig_time setting

### PR DESCRIPTION
**Pull request by ryvnf**
#### Goal of the PR

Provide better accessibility for people who accidentally dig instant breakable nodes but want dig repetition and cannot use the `safe_dig_and_place` setting.

This PR solves this problem by introducing a `repeat_dig_interval` which is like `repeat_place_interval` but for digging. The setting simply ensures that it will take at minimum the settings value from finishing digging one node to starting to dig the next node.

#### Does it resolve any reported issue?

Yes, #13726.

## To do

This PR is ready to review.

## How to test

Hop into a world and hold the dig button on nodes which are instant breakable. Could be done by hopping into a creative world in Mineclonia and start digging any nodes.

Try it with the default value and verify that the digging speed remains the same as previously. Then go into the settings menu and update the _Dig repetition interval_ setting to a higher value. Hop into the world again and dig nodes, verify that there is a larger delay between digging each node.

## Things to consider

The range of values allowed is probably something which should be discussed. Previously, the hardcoded value was 0.15. Currently I made the minimum setting value 0.15 and the maximum 2.0 with the default being 0.15.

The `repeat_place_interval` is minimum 0.16 and maximum 2.0. It could be argued that maybe its minimum value should be lowered to 0.15 for consistency with the `repeat_dig_interval` setting.

But perhaps the minimum value of `repeat_dig_inverval` could be lower?

### CODE REVIEW

**Review by grorp — CHANGES_REQUESTED** [review]
> Try it with the default value and verify that the digging speed remains the same as previously.

The existing minimum value of `0.15` for `runData.nodig_delay_timer` only applies for nodes that take zero time to dig. There can be lower values in other cases:

- In Minetest Game creative mode, the most common value seems to be `0.0328125`.
- When digging dirt in Minetest Game without creative mode, the value is `0.14`.

This means that this PR actually (by default) results in a lower digging speed in some cases. To preserve the existing behavior, the default value for `repeat_dig_time` should be `0.0`.

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -116,6 +116,12 @@ always_fly_fast (Always fly fast) bool true
 #    Requires: keyboard_mouse
 repeat_place_time (Place repetition interval) float 0.25 0.16 2
 
+#    The minimum time in seconds it takes between repeated node digging when
+#    holding the dig button.
+#
+#    Requires: keyboard_mouse
+repeat_dig_time (Dig repetition interval) float 0.15 0.15 2
```
Comment (grorp) [inline builtin/settingtypes.txt:123]:
```suggestion
#    The minimum time in seconds it takes between repeated node digging when
#    holding the dig button.
repeat_dig_time (Dig repetition interval) float 0.15 0.15 2.0
```

This setting works (and is very useful!) on touchscreen devices too.

File: minetest.conf.example [hunk 2]
```diff
@@ -50,6 +50,11 @@
 #    type: float min: 0.16 max: 2
 # repeat_place_time = 0.25
 
+#    The minimum time in seconds it takes between digging nodes when holding
+#    the dig button.
+#    type: float min: 0.15 max: 2
+# repeat_dig_time = 0.15
+
```
Comment (grorp) [inline minetest.conf.example:57]: No need to update minetest.conf.example, it'll be regenerated before release.

File: src/client/game.cpp [hunk 3]
```diff
@@ -4427,7 +4430,7 @@ void Game::readSettings()
 	m_cache_enable_fog                   = g_settings->getBool("enable_fog");
 	m_cache_mouse_sensitivity            = g_settings->getFloat("mouse_sensitivity", 0.001f, 10.0f);
 	m_cache_joystick_frustum_sensitivity = std::max(g_settings->getFloat("joystick_frustum_sensitivity"), 0.001f);
-	m_repeat_place_time                  = g_settings->getFloat("repeat_place_time", 0.16f, 2.0);
+	m_repeat_place_time                  = g_settings->getFloat("repeat_place_time", 0.15f, 2.0f);
```
Comment (grorp) [inline src/client/game.cpp:4433]: I suppose `repeat_dig_time` be cached here too.

**Review by grorp — COMMENTED** [review]
Tested a bit, seems good.

I'm not sure what to think of the change to the minimum of `repeat_dig_time`, but it probably won't make a big difference.

**Review by grorp — CHANGES_REQUESTED** [review]
Calls to `g_settings->(de)registerChangedCallback` are missing for the new setting.

**Review by grorp — APPROVED** [review]

**Comment by AsumFace** [comment]
Hi,
we have intentionally extremely fast tools on the servers Tunnelers' Abyss and Centeria and this change is breaking them due to the imposed minimum duration of 0.15 s. We do no see any necessity for breakage and suggest lowering the minimum to 0 in order not to break games that are relying on it.

**Comment by Desour** [comment]
> Hi, we have intentionally extremely fast tools on the servers Tunnelers' Abyss and Centeria and this change is breaking them due to the imposed minimum duration of 0.15 s. We do no see any necessity for breakage and suggest lowering the minimum to 0 in order not to break games that are relying on it.

IIRC, I tested this because I thought this would be an issue. I've used the devtest `basetools:pick_mese` for this. But I couldn't reach instantaneously fast dig times, and concluded it was already broken before the PR, and hence I remained silent.
It's also very possible that I tested wrongly. I'd like to see reproduction steps.
If there's a regression, please open an issue.

**Comment by superfloh247** [comment]
Hard to measure this correctly, but since this is merge into main the 5.9.dev latest master the alien tools pick axe on Wunderwelt feels slower.
I manually patched my client to allow a minimum of 0.025 and the alien tools pick axe feels normal again.

How to measure this properly?

**Pull request merged**